commands/issue.go

```diff
@@ -618,6 +635,97 @@ text is the title and the rest is the description.`, project))
 	messageBuilder.Cleanup()
 }
 
+func editIssue(cmd *Command, args *Args) {
+	issueNumber := 0
+	if args.ParamsSize() > 0 {
+		issueNumber, _ = strconv.Atoi(args.GetParam(0))
+	}
+	if issueNumber == 0 {
+		utils.Check(cmd.UsageError(""))
+	}
+
+	localRepo, err := github.LocalRepo()
+	utils.Check(err)
+
+	project, err := localRepo.MainProject()
+	utils.Check(err)
+
+	gh := github.NewClient(project.Host)
+
+	issue, err := gh.FetchIssue(project, strconv.Itoa(issueNumber))
+
+	utils.Check(err)
+
+	messageBuilder := &github.MessageBuilder{
+		Filename: "ISSUE_EDITMSG",
+		Title:    "issue",
+	}
+
+	messageBuilder.Message = fmt.Sprintf(`%s
+
+%s`, issue.Title, issue.Body)
+
+	messageBuilder.AddCommentedSection(fmt.Sprintf(`Editing issue %d for %s
+
+Update the message for this issue. The first block of
+text is the title and the rest is the description.`, issue.Number, project))
+
+	updateIssueMessage := true
+	flagIssueEdit := args.Flag.Bool("--edit")
+	messageBuilder.Edit = flagIssueEdit
+	flagIssueMessage := args.Flag.AllValues("--message")
+	if len(flagIssueMessage) > 0 {
+		messageBuilder.Message = strings.Join(flagIssueMessage, "\n\n")
+	} else if args.Flag.HasReceived("--file") {
+		messageBuilder.Message, err = msgFromFile(args.Flag.Value("--file"))
+		utils.Check(err)
+	} else {
+		updateIssueMessage = false
+	}
+
+	params := map[string]interface{}{}
+
+	if updateIssueMessage || flagIssueEdit {
+		title, body, err := messageBuilder.Extract()
+		utils.Check(err)
+		if title == "" {
+			utils.Check(fmt.Errorf("Aborting creation due to empty issue title"))
+		}
+		params["title"] = title
+		params["body"] = body
+	}
+
+	flagIssueLabels := commaSeparated(args.Flag.AllValues("--labels"))
+	if len(flagIssueLabels) > 0 {
+		params["labels"] = flagIssueLabels
+	}
+
+	flagIssueAssignees := commaSeparated(args.Flag.AllValues("--assign"))
+	if len(flagIssueAssignees) > 0 {
+		params["assignees"] = flagIssueAssignees
+	}
+
+	milestoneNumber, err := milestoneValueToNumber(args.Flag.Value("--milestone"), gh, project)
+	utils.Check(err)
+	if milestoneNumber > 0 {
+		params["milestone"] = milestoneNumber
+	}
```

Comment:
These flags could be moved into shared functions that could be called from edit and create. That might add additional indirection and I would personally keep it the way it is. Let me know what you think.